memor/session.py

```diff
@@ -51,49 +48,38 @@ def __init__(
             if messages:
                 self.update_messages(messages)
 
-    def __eq__(self, other_session):
+    def __eq__(self, other_session: Any) -> bool:
         """
         Check sessions equality.
 
         :param other_session: other session
-        :type other_session: Session
-        :return: bool
         """
         if isinstance(other_session, Session):
             return self._title == other_session._title and self._messages == other_session._messages
         return False
 
-    def __str__(self):
+    def __str__(self) -> str:
         """Return string representation of Session."""
         return self.render(render_format=RenderFormat.STRING)
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         """Return string representation of Session."""
         return "Session(title={title})".format(title=self._title)
 
-    def __len__(self):
-        """
-        Return the length of the Session object.
-
-        :return: length of the Session object
-        """
+    def __len__(self) -> int:
+        """Return the length of the Session object."""
         return len(self._messages)
 
-    def __iter__(self):
-        """
-        Iterate through the Session object.
 
-        :return: message as Generator[Prompt/Response]
-        """
+    def __iter__(self) -> Generator[Union[Prompt, Response], None, None]:
```

Comment:
It appears that `Generator[Union[Prompt, Response], None, None]` is incorrect. The correct version should be `Generator[Union[Prompt, Response]]`?

Reply:
I used defintions from [here](https://stackoverflow.com/questions/42531143/how-to-type-hint-a-generator-in-python-3).